pkg/identityserver/end_device_registry.go

```diff
@@ -283,6 +342,10 @@ func (is *IdentityServer) updateEndDevice(ctx context.Context, req *ttnpb.Update
 		req.FieldMask = cleanFieldMaskPaths(ttnpb.EndDeviceFieldPathsNested, req.FieldMask, nil, []string{"activated_at"})
 	}
 
+	if ttnpb.HasAnyField(ttnpb.TopLevelFields(req.FieldMask.GetPaths()), "claim_authentication_code") {
+		req.FieldMask.Paths = append(req.FieldMask.GetPaths(), "claim_authentication_code")
+	}
```

Comment:
Idem; what is this for?

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/blob/0c2ff932c25f454e9d24ef83322686edf2a2c83e/pkg/identityserver/gormstore/end_device.go#L228
https://github.com/TheThingsNetwork/lorawan-stack/blob/0c2ff932c25f454e9d24ef83322686edf2a2c83e/pkg/identityserver/gormstore/end_device.go#L152

The getter/setter is only called if the paths contain the top-level field `claim_authentication_code`.
The CLI only sets the sub-fields in the fieldmask (ex: `claim_authentication_code.value`). 
So ,in the registry, we check if the sub-fields are set to set the top level field.
This is the same as the implementation for the `picture` field in `L350-L352`.
(PS: This is also applicable for the bun store since it uses `ttnpb.SetFields`).